daemon/daemon.go

```diff
@@ -1221,16 +1221,16 @@ func (daemon *Daemon) verifyHostConfig(hostConfig *runconfig.HostConfig) ([]stri
 }
 
 func (daemon *Daemon) setHostConfig(container *Container, hostConfig *runconfig.HostConfig) error {
-	if err := daemon.registerMountPoints(container, hostConfig); err != nil {
-		return err
-	}
-
 	container.Lock()
 	defer container.Unlock()
 	if err := parseSecurityOpt(container, hostConfig); err != nil {
 		return err
 	}
 
+	if err := daemon.registerMountPoints(container, hostConfig); err != nil {
```

Comment:
@jfrazelle https://github.com/cpuguy83/docker/commit/e83230bbd34946dc7189b363d4ce95c786b14dc1 to fix locking